cosmos/dbt/graph.py

```diff
@@ -424,18 +428,21 @@ def load_via_dbt_ls_without_cache(self) -> None:
             raise CosmosLoadDbtException("Unable to load project via dbt ls without a profile config.")
 
         with tempfile.TemporaryDirectory() as tmpdir:
-            logger.debug(
-                f"Content of the dbt project dir {self.render_config.project_path}: `{os.listdir(self.render_config.project_path)}`"
-            )
+            logger.debug(f"Content of the dbt project dir {project_path}: `{os.listdir(project_path)}`")
             tmpdir_path = Path(tmpdir)
 
             create_symlinks(project_path, tmpdir_path, self.render_config.dbt_deps)
 
-            if self.project.partial_parse and self.cache_dir:
-                latest_partial_parse = cache._get_latest_partial_parse(project_path, self.cache_dir)
+            latest_partial_parse = None
+            if self.project.partial_parse:
+                if self.should_use_partial_parse_cache() and self.cache_dir:
```

Comment:
Do we still need check `self.cache_dir` here, considering that `should_use_partial_parse_cache` already performs check `bool(self.cache_dir)`?

Reply:
+1

Reply:
@pankajastro @pankajkoti MyPy didn't like it without it :( That's why I added it there. Do you have any suggestions

Reply:
okay looks like MyPy is not able to figure out what's happening in the previous method. So I guess we can keep it for now until find a better resolution if any.